sdk/keyvault/keyvault-keys/test/tracing.test.ts

```diff
@@ -0,0 +1,245 @@
+      ),
+      dq: toBytes(
+        "00a0bd4c0a3d9f64436a082374b5caf2488bac1568696153a6a5e4cd85d186db31e2f58f024c617d29f37b4e6b54c97a1e25efec59c4d1fd3061ac33509ce8cae5c11f4cd2e83f41a8264f785e78dc0996076ee23dfdfc43d67c463afaa0180c4a718357f9a6f270d542479a0f213870e661fb950abca4a14ca290570ba7983347"
+      ),
+      qi: toBytes(
+        "009fe7ae42e92bc04fcd5780464bd21d0c8ac0c599f9af020fde6ab0a7e7d1d39902f5d8fb6c614184c4c1b103fb46e94cd10a6c8a40f9991a1f28269f326435b6c50276fda6493353c650a833f724d80c7d522ba16c79f0eb61f672736b68fb8be3243d10943c4ab7028d09e76cfb5892222e38bc4d35585bf35a88cd68c73b07"
+      )
+    };
+    await client.importKey(keyName, jsonWebKey, { tracingOptions });
+  });
+
+  testTracingNodeNames(["createEcKey", "updateKeyProperties"], async function(
+    tracingOptions: OperationTracingOptions
+  ) {
+    const keyName = getKeyName(this);
+    const { version } = (await client.createEcKey(keyName, { tracingOptions })).properties;
+    await client.updateKeyProperties(keyName, version || "", { enabled: false, tracingOptions });
+    await testClient.flushKey(keyName);
+  });
+
+  testTracingNodeNames(
+    ["listPropertiesOfKeys", "listPropertiesOfKeyVersions", "listDeletedKeys"],
+    async function(tracingOptions: OperationTracingOptions) {
+      const keyName = testClient.formatName(`${keyPrefix}-${this!.test!.title}-${keySuffix}`);
+      const keyNames = [`${keyName}-0`, `${keyName}-1`];
+      for (const name of keyNames) {
+        await client.createKey(name, "RSA");
+      }
+
+      for await (const properties of client.listPropertiesOfKeys({ tracingOptions })) {
+        for await (const _version of client.listPropertiesOfKeyVersions(properties.name, {
+          tracingOptions
+        })) {
+          // Nothing to do here..
+        }
+      }
+
+      for (const name of keyNames) {
+        const poller = await client.beginDeleteKey(name, testPollerProperties);
+        await poller.pollUntilDone();
+      }
+
+      for await (const _deletedKey of client.listDeletedKeys({ tracingOptions })) {
+        // Nothing to do here..
+      }
+
+      for (const name of keyNames) {
+        await testClient.purgeKey(name);
+      }
+    }
+  );
+
+  if (isRecordMode() || isPlaybackMode()) {
+    // This test can't run live,
+    // since the purge operation currently can't be expected to finish anytime soon.
+    testTracingNodeNames(["createRsaKey", "backupKey", "restoreKeyBackup"], async function(tracingOptions: OperationTracingOptions) {
+      const keyName = getKeyName(this);
+      await client.createRsaKey(keyName, { tracingOptions });
+      const backup = await client.backupKey(keyName, { tracingOptions });
+
+      await testClient.flushKey(keyName);
+
+      // This method can't be made into a long running operation yet.
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```

Comment:
why an infinite loop rather than a promise that can be awaited?

Reply:
if you pull it out into a function it can await itself recursively

Reply:
We had a retry mechanism before, but I feel it only obscures what is clearly wrong. These methods where we do these while tests need to be LROs.

In some cases we can't implement them (like in purging). But I believe I can do LROs for restoring stuff, since it's completed once it is effectively retrievable. I'll make an issue to implement LROs for recover operations.

Reply:
I've made a new issue: https://github.com/Azure/azure-sdk-for-js/issues/8934